pallets/providers/README.md

```diff
@@ -1 +1,555 @@
+
+- `add_bucket`, which adds a new entry to the map.
+- `change_root_bucket`, which changes the corresponding bucket's root.
+- `remove_root_bucket`, which removes the entry of the corresponding bucket.
+
+#### Fields
+
+It's a map from a bucket ID to that bucket's metadata
+
+```
+BucketId -> Bucket
+```
+
+### `AccountIdToBackupStorageProviderId`
+
+This storage is used to keep track of the one-to-one relationship between an account ID and a Backup Storage Provider ID, which is used to both choose which challenges are requested from that Storage Provider and to compare with the threshold used to allow Backup Storage Providers to offer themselves to store a new file of the system. It is updated in:
+
+- `confirm_sign_up`, which adds a new entry to the map if the account to confirm is a Backup Storage Provider.
+- `bsp_sign_off`, which removes the corresponding entry from the map.
+
+#### Fields
+
+It's a map from an account ID to a Backup Storage Provider ID, which is of the Hash type from the runtime.
+
+```
+AccountId -> BackupStorageProviderId
+```
+
+### `BackupStorageProviders`
+
+This storage holds the metadata of each registered Backup Storage Provider, which has its capacity, its used data, the valid multiaddresses to connect to it, its forest root and the block in which this Storage Provider last changed its capacity. It is updated in:
+
+- `confirm_sign_up`, which adds a new entry to the map if the account to confirm is a Backup Storage Provider.
+- `bsp_sign_off`, which removes the corresponding entry from the map.
+- `change_capacity`, which changes the entry's `capacity`.
+
+#### Fields
+
+It's a map from a Backup Storage Provider ID to its metadata.
+
+```
+BackupStorageProviderId -> BackupStorageProviderMetadata
+```
+
+### `MspCount`
+
+This storage holds the amount of Main Storage Providers that are currently registered in the system. It is updated in:
+
+- `confirm_sign_up`, which adds one to this storage if the account to confirm is a Main Storage Provider.
+- `msp_sign_off`, which subtracts one from this storage.
+
+### `BspCount`
+
+This storage holds the amount of Backup Storage Providers that are currently registered in the system. It is updated in:
+
+- `confirm_sign_up`, which adds one to this storage if the account to confirm is a Backup Storage Provider.
+- `bsp_sign_off`, which subtracts one from this storage.
+
+### `TotalBspsCapacity`
+
+This storage holds the sum of all the capacity that has been registered by Backup Storage Providers, which corresponds to the capacity of the whole network. It is updated in:
+
+- `confirm_sign_up`, which adds the capacity of the registered Storage Provider to this storage if the account to confirm is a Backup Storage Provider.
+- `bsp_sign_off`, which subtracts the capacity of the Backup Storage Provider to sign off from this storage.
+
+## Events
```

Comment:
I know I am being a bummer, but I would explain the events at a high level and how normally actors in the network would react to them.